apps/wallet/src/engine/runners/solana/history.ts

```diff
@@ -0,0 +1,318 @@
+import type {
+	Connection,
+	ParsedInstruction,
+	ParsedTransaction,
+	ParsedTransactionMeta,
+	ParsedTransactionWithMeta,
+	PublicKey,
+	TokenBalance,
+} from '@solana/web3.js';
+import type { TransactionHistory } from '@walless/core';
+import { Networks } from '@walless/core';
+import { solana } from '@walless/network';
+import type { TokenDocument, TransactionHistoryDocument } from '@walless/store';
+import { selectors } from '@walless/store';
+import { storage } from 'utils/storage';
+
+import { getMetadata } from './metadata';
+import type { ParsedTokenAccountWithAddress } from './types';
+
+const historyLimit = 20;
+
+export const getTransactionsHistory = async (
+	connection: Connection,
+	wallet: PublicKey,
+	accounts: ParsedTokenAccountWithAddress[],
+) => {
+	const splTokenAccount = accounts
+		.filter((account) => account.tokenAmount.decimals > 0)
+		.map((account) => account.publicKey);
+
+	const accountList = [wallet].concat(splTokenAccount);
```

Comment:
It's good to use built-in array method, but It looks weird because `concat` is usually used to merge two arrays into a new array (we can also do this by `[...a, ...b]`. In this case, you constructed a new array to merge, so the good one for this is using `push`. `splTokenAccounts.push(wallet)`, and if pushing the wallet account makes `splTokenAccounts` become non-sense, you should rename it just `tokenAccounts` I guess